### PR TITLE
Refactor Mailchimp recipients and segmentation options

### LINKED ISSUE #22: Support For Mailchimp Segments
Add support for Mailchimp segments: https://us7.admin.mailchimp.com/lists/members?id=453669&previewName=Contacts#p:1-s:10-sa:last_update_time-so:false

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PRs proposes a change on how we support segmentation with the Mailchimp provider. Currently, we have support for selecting a group and multiple tags. The tags implementation also included saved segments, although not supported due to the segmentation options for `saved` segments being different for `static` segments (tags).

The redesign fixes the segmentation support and proposes a simpler approach, similar to how Mailchimp's wizard works, in which you are able to select one of the available groups, tags, or segments. For what we generally propose in terms of form complexity, this approach avoids scenarios in which the user would select a group and different segments that don't match existing recipients.

It also removes the `Use Mailchimp tags` field. It looks like an unnecessary block for accessing segmentation options.

image

image

Closes #22.

### How to test the changes in this Pull Request:

1. Switch to this branch and run `npm run build`
2. Test selecting different segmentation options and publishing campaigns

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php [hunk 1]
```diff
@@ -561,6 +562,14 @@ public function audience_segments( $post_id, $compound_interest_id, $tag_ids ) {
 				__( 'Mailchimp campaign ID not found.', 'newspack-newsletters' )
 			);
 		}
+
+		if ( '' !== $target_id && ! $interest_id && ! $segment_id ) {
```
Comment (adekbadek) [inline includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php:566]: If the condition on line 550 is not met, `$interest_id` will not be defined. It should either be defined before the condition, or its existence checked here.

File: src/service-providers/mailchimp/ProviderSidebar.js [hunk 2]
```diff
@@ -3,75 +3,71 @@
...
+	}
 
 	return (
 		<Fragment>
-			{ availableInterests.length ? (
+			{ options.length ? (
 				<SelectControl
-					label={ __( 'Groups', 'newspack-newsletters' ) }
-					value={ segmentsData.interest_id }
+					label={ __( 'Group, segment or tag', 'newspack-newsletters' ) }
```
Comment (adekbadek) [inline src/service-providers/mailchimp/ProviderSidebar.js:61]: Oxford-comma nitpick ;) : 
```suggestion
					label={ __( 'Group, segment, or tag', 'newspack-newsletters' ) }
```

**Review by adekbadek — CHANGES_REQUESTED** [review]
Works well! Just two nitpicks.

Also, perhaps for a later PR - the pre-send modal could indicate that a segment or group was chosen. Here, I know my list has more contacts than 1 (but with the chosen segment the newsletter will be sent to just one contact), so it's weird to see `1 subscriber`:

image

File: includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php (hunk 1 above)
Comment (miguelpeixe) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php:569]: Nice catch! 8aba70be868cc7d36a43c1c05786838e9a95d8f5 fixes it.

**Comment by miguelpeixe** [comment]
> Also, perhaps for a later PR - the pre-send modal could indicate that a segment or group was chosen. Here, I know my list has more contacts than 1 (but with the chosen segment the newsletter will be sent to just one contact), so it's weird to see `1 subscriber`

That would certainly be a nice addition, I filed #558 for this.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**